airflow/utils/dag_processing.py

```diff
@@ -44,101 +45,13 @@
 # To avoid circular imports
 import airflow.models
 from airflow import configuration as conf
-from airflow.dag.base_dag import BaseDag, BaseDagBag
-from airflow.exceptions import AirflowException
 from airflow.settings import logging_class_path
 from airflow.utils import timezone
 from airflow.utils.db import provide_session
 from airflow.utils.log.logging_mixin import LoggingMixin
 from airflow.utils.state import State
 
 
-class SimpleDag(BaseDag):
```

Comment:
Can you explain your thinking behind removing this class?

Reply:
The SimpleDag is almost an exact copy of the dag, with the addition that it is serializable. I had a discussion with Peter, and I fully agree with him that the additional complexity of having this class hierarchy (`SimpleDag`, `BaseDag`, `Dag`) isn't worth the upsides that it brings. 

Firstly, my question would be, can we make the `Dag` in such a way that we can serialize this one as well. Secondly, why do we serialize in general? Afaik it was because of performance reasons (the webserver shows the dag based on the pickled instance), but I'm in doubt if we don't parse all the dags all the time if we still need this mechanism.